app/models/task.py

```diff
@@ -1,6 +1,33 @@
 from flask import current_app
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+
+    COLUMNS = ["title", "description", "completed_at", "goal_id"]
+    
+    def to_dict(self):
+        is_complete = False if self.completed_at == None else True
+        if self.goal_id:
+            return {
+                "id": self.task_id,
+                "goal_id": self.goal_id,
+                "title" : self.title,
+                "description" : self.description,
+                "is_complete" : is_complete
+            }   
+        else:
+            return {
+                "id": self.task_id,
+                "title" : self.title,
+                "description" : self.description,
+                "is_complete" : is_complete
+            }    
```

Comment:
Great helper method Vange! Notice how the dictionaries in the if/else are very similar? We can refactor by making a 'base' dictionary to return as is or if there's an existing goal id:

```suggestion
    def to_dict(self):
        is_complete = False if self.completed_at == None else True
        
        task_dict = {
                "id": self.task_id,
                "title" : self.title,
                "description" : self.description,
                "is_complete" : is_complete
            }   
            
        if self.goal_id:
            task_dict["goal_id"] = self.goal_id
            return task_dict 
        else:
            return task_dict
```